### PR TITLE
Server improvements

### PR DESCRIPTION
**Pull request by kompotkot**
Added and modified:
- logging;
- CORS and panic middleware;
- `Validate()` logs directly to client to improve error messaging;
- `address` for `authorize` CLI changed to `source-id` to have "sameness" with server endpoints;
- `token-id` for `authorize` CLI changed to `source-token-id` to have "sameness" with server endpoints;
- updated READMEs with basic workflow commands

### CODE REVIEW

File: relayers/erc721_relayer.go [hunk 1]
```diff
@@ -174,42 +179,39 @@ func (relayer *ERC721Relayer) Validate(recipient common.Address, tokenID, source
...
 
 	owner, callErr := contract.OwnerOf(callOpts, sourceTokenID)
 	if callErr != nil {
-		return false, callErr
+		log.Printf("contract owner check error, err: %s", callErr.Error())
+		return errors.New("Internal server error")
+	}
+
+	if signerAddress != owner {
+		return errors.New("Token owner not verified")
```
Comment (kompotkot) [inline relayers/erc721_relayer.go:204]: Message signer is not token owner

File: relayers/erc721_relayer.go [hunk 2]
```diff
@@ -129,41 +129,46 @@ func (relayer *ERC721Relayer) Address() (common.Address, error) {
 	return relayer.address, nil
 }
 
-func (relayer *ERC721Relayer) Validate(recipient common.Address, tokenID, sourceID, sourceTokenID, liveUntil *big.Int, metadataURI string, authorizationMessage interface{}) (bool, error) {
+// Validate handles signature validation.
+// Important! All errors going to client response, do not pass signatures and valuable information.
+func (relayer *ERC721Relayer) Validate(recipient common.Address, tokenID, sourceID, sourceTokenID, liveUntil *big.Int, metadataURI string, authorizationMessage interface{}) error {
```
Comment (kompotkot) [inline relayers/erc721_relayer.go:134]: Define list of errors and response only specified error to user like:
`var ErrUnauthorizedRequest error = errors.New("unauthorized request")`

**Review by kompotkot — COMMENTED** [review]
comments

**Review by zomglings — COMMENTED** [review]
lgtm

**Review by zomglings — APPROVED** [review]
lgtm

**Pull request merged**